### PR TITLE
fix: implements the nodeSelector check

### PR DESCRIPTION
**Pull request by nbalacha**
This commit implements the nodeSelector check in
vgmanager.

Signed-off-by: N Balachandran <nibalach@redhat.com>

### CODE REVIEW

File: pkg/vgmanager/vgmanager_controller.go [hunk 1]
```diff
@@ -90,12 +91,11 @@ func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 		return ctrl.Result{}, err
 	}
 
-	//TODO: actually check the node against the nodeSelector.
-	node := &corev1.Node{}
-	nodeMatches, err := NodeSelectorMatchesNodeLabels(node, volumeGroup.Spec.NodeSelector)
+	//Does this node need to process it?
```
Comment (sp98) [inline pkg/vgmanager/vgmanager_controller.go:94]: can you elaborate a bit more?

File: pkg/vgmanager/vgmanager_controller.go [hunk 2]
```diff
@@ -381,3 +381,21 @@ func (r *VGReconciler) getNewNodeStatus(status *lvmv1alpha1.VGStatus) *lvmv1alph
...
+	err := r.Client.Get(ctx, types.NamespacedName{Name: r.NodeName}, node)
+	if err != nil {
+		return false, err
+	}
+	nodeMatches, err := NodeSelectorMatchesNodeLabels(node, selector)
+	if err != nil {
+		return false, err
+	}
+
+	if !nodeMatches {
```
Comment (sp98) [inline pkg/vgmanager/vgmanager_controller.go:397]: This check might not be needed. Function can return `nodeMatches` variable.

File: pkg/vgmanager/vgmanager_controller.go (hunk 1 above)
Comment (nbalacha) [reply pkg/vgmanager/vgmanager_controller.go:94]: If the nodeSelector for the VG does not match the node the vgmanager is running on, it does not need to do anything.

File: pkg/vgmanager/vgmanager_controller.go (hunk 2 above)
Comment (nbalacha) [reply pkg/vgmanager/vgmanager_controller.go]: done.

File: pkg/vgmanager/vgmanager_controller.go (hunk 1 above)
Comment (sp98) [reply pkg/vgmanager/vgmanager_controller.go]: ok.  The comment is framed as a question.  Should it rather be a statement.

File: pkg/vgmanager/vgmanager_controller.go [hunk 3]
```diff
@@ -381,3 +382,14 @@ func (r *VGReconciler) getNewNodeStatus(status *lvmv1alpha1.VGStatus) *lvmv1alph
...
 }
+
+func (r *VGReconciler) matchesMyNode(ctx context.Context, selector *corev1.NodeSelector) (bool, error) {
+
+	node := &corev1.Node{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: r.NodeName}, node)
+	if err != nil {
+		return false, err
+	}
+	nodeMatches, err := NodeSelectorMatchesNodeLabels(node, selector)
```
Comment (sp98) [inline pkg/vgmanager/vgmanager_controller.go:393]: maybe just do 
` return NodeSelectorMatchesNodeLabels(node, selector)`

File: pkg/vgmanager/vgmanager_controller.go [hunk 4]
```diff
@@ -381,3 +382,14 @@ func (r *VGReconciler) getNewNodeStatus(status *lvmv1alpha1.VGStatus) *lvmv1alph
 	setStatus(status, vgNodeStatus)
 	return vgNodeStatus
 }
+
+func (r *VGReconciler) matchesMyNode(ctx context.Context, selector *corev1.NodeSelector) (bool, error) {
```
Comment (sp98) [inline pkg/vgmanager/vgmanager_controller.go:386]: need a better name than `matchesMynode`.

Comment (leelavg) [inline pkg/vgmanager/vgmanager_controller.go:386]: - maybe we'll not be using this function again and considering it's quite small, how about in-lining it in the call site itself?

Comment (nbalacha) [reply pkg/vgmanager/vgmanager_controller.go:386]: I think that is best left to the compiler.

Comment (nbalacha) [reply pkg/vgmanager/vgmanager_controller.go:386]: Suggestions?

Comment (leelavg) [reply pkg/vgmanager/vgmanager_controller.go:386]: ah! got it.

Comment (sp98) [reply pkg/vgmanager/vgmanager_controller.go:386]: I also agree with Leela here. We can remove the `matchesMyNode` function.

Comment (sp98) [reply pkg/vgmanager/vgmanager_controller.go:386]: `matchesNode`.   But I also agree with Leela's comment that we can remove this function.

Comment (nbalacha) [reply pkg/vgmanager/vgmanager_controller.go:386]: I prefer having functions for readability.

File: pkg/vgmanager/vgmanager_controller.go (hunk 3 above)
Comment (nbalacha) [reply pkg/vgmanager/vgmanager_controller.go]: done

File: pkg/vgmanager/vgmanager_controller.go (hunk 4 above)
Comment (nbalacha) [reply pkg/vgmanager/vgmanager_controller.go]: I disagree - functions provide readability.

**Review by leelavg — APPROVED** [review]
/lgtm

**Pull request merged**